Homework_6/homework_6.js

```diff
@@ -0,0 +1,117 @@
+
+function getEmployeeInfo(name) {
+    let employeeNames = ['Artsiom', 'Nikolai', 'Antanina', 'Dani', 'Antony']
+    let employeeSalary = [2000, 3000, 4000, 5000, 6000]
+    let result = []
+
+    for (let i = 0; i < employeeNames.length; i++) {
+      
+        if (name === employeeNames[i]) {
+            result.push(employeeNames[i])
+            result.push(employeeSalary[i])
+            return result
+        }
+    }
+
+    return null
+}
+
+getEmployeeInfo('Dani')
+console.log(getEmployeeInfo('Dani'))
+
+
+
+
+// Task_2
+// 1. У вас есть массив названий пицц вашего конкурента. Создайте функцию, которая будет принимать ваш набор названий пицц (массив) 
+//   и возвращать только те, которых нет у конкурента (тоже массив). Если все ваши пиццы есть у конкурента - вернуть null
+//   Пиццы конкурента:
+//   const competitorPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai']
+
+function getUniquePizzaName(name) {
+    const competitorPizzas = ['Peperoni', 'Caprichosa', 'Diablo', '4 cheeses', 'hawai']
+    const result = []
+
+    for (let i = 0; i < name.length; i++) {
+        
+        if (!competitorPizzas.includes(name[i])) {
+            result.push(name[i])
+        }
+
+    }
+
+    if (result.length == 0) {
+        return null
+    } else {
+        return result
+    }
+}
+
+console.log(getUniquePizzaName(['Peperoni', 'Diablo', '4 cheeses', 'hawai', '111', '4 444',]))
+
+// 2. Написать функцию, которая принимает предложение (слова разделенные только пробелами) в качестве параметра и выводит в консоль слово с наибольшим количеством букв. 
+//   Если таких слов несколько - выводит их все.
+
+function getWordWithMaxCharacters(sentense) {
+    const arr = sentense.split(' ')
+    let counter = 0
+    let newArr = []
+
+    for (i = 0; i < arr.length; i++) {
+
+        if (arr[i].length > counter) {
+            counter = arr[i].length
+        }
+
+    }
+
+    for(let i = 0; i < arr.length; i++){
+
+        if (arr[i].length === counter) {
+            newArr.push(arr[i])
+        }
+
+    }
+
+    console.log(newArr.join(' '))
+}
+
+getWordWithMaxCharacters('and home and')
+
+// 3. Напишите функцию, которая принимает на вход массив чисел, убирает из него дубликаты и возвращает массив с только уникальными значениями.
+
+function deleteDuplicates(array) {
+    let result = []
+
+    for (let str of array) {
+      if (!result.includes(str)) {
+        result.push(str);
+      }
+    }
+  
+    return result;
+  }
+
+
+// 4. Написать функцию, которая принимает на вход слово и проверяет, является ли это слово палиндромом
+
+function checkIfWordIsPalindrome(word) {
+    
+    if (typeof word !== 'string') {
+        console.log('Input is not a string')
+    } else if (word === word.split('').reverse().join('')) {
+        console.log('Palindrome')
+    } else {
+        console.log('No')
+    }
+}
+console.log(checkIfWordIsPalindrome(123))  //do not understand why 'undefined' is displayed in output after console.log
```

Comment:
```suggestion
console.log(checkIfWordIsPalindrome(123))  //do not understand why 'undefined' is displayed in output after console.log
```

Because you need to return result of the function, not just use `console.log` in function body
`In JavaScript, if a function does not explicitly return a value, it returns undefined by default.`
*When you call console.log(checkIfWordIsPalindrome(123)), the function runs, prints either "Input is not a string", "Palindrome", or "No" to the console and then returns `undefined`. The console.log statement then prints this `undefined` value.*